File: dotnet/Lexy/Lexy.Compiler/Language/Comments.cs
using System.Collections.Generic;
using Lexy.Compiler.Parser;
using Lexy.Compiler.Parser.Tokens;

namespace Lexy.Compiler.Language
{
    public class Comments : ParsableNode
    {
        private IList<string> lines = new List<string>();

        public Comments(SourceReference sourceReference) : base(sourceReference)
        {
        }

        public override IParsableNode Parse(IParserContext context)
        {
            var valid = context.ValidateTokens<Comments>()
                .Count(1)
                .Comment(0)
                .IsValid;

            if (!valid) return null;

            var comment = context.CurrentLine.Tokens.Token<CommentToken>(0);
            lines.Add(comment.Value);
            return this;
        }

        public override IEnumerable<INode> GetChildren()
        {
            yield break;
        }

        protected override void Validate(IValidationContext context)
        {
        }
    }
}

File: typescipt/src/language/comments.ts


export class Comments extends ParsableNode {
   private readonly Array<string> lines = list<string>(): new;

   public Comments(SourceReference sourceReference) : base(sourceReference) {
   }

   public override parse(context: IParseLineContext): IParsableNode {
     let valid = context.ValidateTokens<Comments>()
       .Count(1)
       .Comment(0)
       .IsValid;

     if (!valid) return null;

     let comment = context.Line.Tokens.Token<CommentToken>(0);
     lines.Add(comment.Value);
     return this;
   }

   public override getChildren(): Array<INode> {
     yield break;
   }

   protected override validate(context: IValidationContext): void {
   }
}
